### PR TITLE
Add workflow for auto deployment to dev for main branch changes

### PR DESCRIPTION
**Pull request by Gavinok**
This is a draft for #429

This will likely need extra review since there isn't an option for local testing in this case. Most of the code has been ported from 

https://github.com/bcgov/traction/blob/restore-json-toggle-schema/.github/workflows/on_push_main.yaml

### CODE REVIEW

**Review by i5okie — COMMENTED** [review]
Have the secrets been added in the repo settings?

**Review by i5okie — DISMISSED** [review]
lgtm

**Comment by Gavinok** [comment]
> Have the secrets been added in the repo settings?

As far as I can tell I don't have the proper permissions to add or see them. Since we have not workflows using them yet they probably need to be added

**Comment by esune** [comment]
@i5okie we added all certificates except for `OPENSHIFT_CA_CRT` - I am not 100% sure what should go there. Can you help with that?
After adding that secret we should be able to merge and test this.

**Review by esune — APPROVED** [review]
👍🏻

File: .github/workflows/on_push_main.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+          namespace: ${{ secrets.OPENSHIFT_NAMESPACE }}
+
+      - name: VC-Authn Dev Helm
+        run: |
+          helm upgrade --install vc-authn-oidc \
+          -f ./charts/vc-authn-oidc/templates/deployment.yaml \
+          --set image.tag="dev" \
+          ./charts/vc-authn-oidc --wait
+      - name: Restart Deployments
+        # TODO determine the vc-authn equivalent
```
Comment (loneil) [inline .github/workflows/on_push_main.yaml:44]: Should the TODO comments stay in?

Comment (esune) [reply .github/workflows/on_push_main.yaml:44]: Ha, good catch! It looks like this is resolved so we should remove it

File: .github/workflows/on_push_main.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      - name: VC-Authn Dev Helm
+        run: |
+          helm upgrade --install vc-authn-oidc \
+          -f ./charts/vc-authn-oidc/templates/deployment.yaml \
+          --set image.tag="dev" \
+          ./charts/vc-authn-oidc --wait
+      - name: Restart Deployments
+        run: |
+          oc rollout restart deployment/vc-authn-oidc
+          oc rollout restart deployment/vc-authn-controller
```
Comment (esune) [inline .github/workflows/on_push_main.yaml:45]: I double checked, we actually need to trigger deployments `vc-authn-oidc-agent` and `vc-authn-oidc`

**Review by esune — CHANGES_REQUESTED** [review]

File: .github/workflows/on_push_main.yaml (hunk 2 above)
Comment (Gavinok) [reply .github/workflows/on_push_main.yaml:45]: Oh, I'll add that now. My mistake, I figured the agent was fine.

File: .github/workflows/on_push_main.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        run: |
+          helm upgrade --install vc-authn-oidc \
+          -f ./charts/vc-authn-oidc/templates/deployment.yaml \
+          --set image.tag="dev" \
+          ./charts/vc-authn-oidc --wait
+      - name: Restart Deployments
+        run: |
+          oc rollout restart deployment/vc-authn-oidc-agent
+          oc rollout restart deployment/vc-authn-oidc
+          oc rollout restart deployment/vc-authn-controller
```
Comment (esune) [inline .github/workflows/on_push_main.yaml:46]: This actually doesn't exist, we should remove it completely

**Review by esune — CHANGES_REQUESTED** [review]

**Pull request merged**